functions/types.go

```diff
@@ -0,0 +1,124 @@
+package functions
+
+import (
+	substraitgo "github.com/substrait-io/substrait-go"
+	"github.com/substrait-io/substrait-go/types"
+	"strings"
+)
+
+var nameToTypeMap = map[string]types.Type{
+	"boolean":       &types.BooleanType{},
+	"i8":            &types.Int8Type{},
+	"i16":           &types.Int16Type{},
+	"i32":           &types.Int32Type{},
+	"i64":           &types.Int64Type{},
+	"fp32":          &types.Float32Type{},
+	"fp64":          &types.Float64Type{},
+	"binary":        &types.BinaryType{},
+	"string":        &types.StringType{},
+	"timestamp":     &types.TimestampType{},
+	"date":          &types.DateType{},
+	"time":          &types.TimeType{},
+	"timestamp_tz":  &types.TimestampTzType{},
+	"interval_year": &types.IntervalYearType{},
+	"interval_day":  &types.IntervalDayType{},
+	"uuid":          &types.UUIDType{},
+
+	// short names
+	"bool":  &types.BooleanType{},
+	"vbin":  &types.BinaryType{},
+	"str":   &types.StringType{},
+	"ts":    &types.TimestampType{},
+	"tstz":  &types.TimestampTzType{},
+	"iyear": &types.IntervalYearType{},
+	"iday":  &types.IntervalDayType{},
+}
+
+func isSupportedType(typeString string) bool {
+	_, ok := nameToTypeMap[typeString]
+	return ok
+}
+
+type typeRegistryImpl struct {
+	typeMap map[string]types.Type
+}
+
+func NewTypeRegistry() TypeRegistry {
+	return &typeRegistryImpl{typeMap: nameToTypeMap}
+}
+
+func (t *typeRegistryImpl) GetTypeFromTypeString(typeString string) (types.Type, error) {
+	return getTypeFromTypeString(typeString, t.typeMap)
+}
+
+func getTypeFromTypeString(typeString string, typeMap map[string]types.Type) (types.Type, error) {
```

Comment:
this should take map[string]typeInfo as argument too

Reply:
As mentioned in [this](https://github.com/substrait-io/substrait-go/pull/32#discussion_r1709041983) comment, it will be resolved in future PR